tailnet/service.go

```diff
@@ -89,20 +91,40 @@ func NewClientService(options ClientServiceOptions) (
 	return s, nil
 }
 
-func (s *ClientService) ServeClient(ctx context.Context, version string, conn net.Conn, id uuid.UUID, agent uuid.UUID) error {
+type ServeClientOptions struct {
+	Peer uuid.UUID
+	// Include for multi-workspace service
+	Database UpdateQuerier
+	// Include for single workspace service
+	Agent *uuid.UUID
+}
+
+func (s *ClientService) ServeClient(ctx context.Context, version string, conn net.Conn, opts ServeClientOptions) error {
```

Comment:
it looks like the opts are just being used to construct the `StreamID`.  Why not just pass the `StreamID` directly.  Less branching because the two different callers of this function know which type of `CoordinateeAuth` they need.